compiler/backend/src/org/jetbrains/kotlin/codegen/inline/SMAP.kt

```diff
@@ -267,72 +214,36 @@ class SMAP(val fileMappings: List<FileMapping>) {
 
 data class SMAPAndMethodNode(val node: MethodNode, val classSMAP: SMAP)
 
-class RawFileMapping(val name: String, val path: String) {
-    private val rangeMappings = arrayListOf<RangeMapping>()
-
-    private var lastMappedWithNewIndex = -1000
-
-    fun toFileMapping() =
-        FileMapping(name, path).apply {
-            for (range in rangeMappings) {
-                addRangeMapping(range)
-            }
-        }
-
-    fun initRange(start: Int, end: Int) {
-        assert(rangeMappings.isEmpty()) { "initRange should only be called for empty mapping" }
-        rangeMappings.add(RangeMapping(start, start, end - start + 1))
-        lastMappedWithNewIndex = end
-    }
+class FileMapping(val name: String, val path: String) {
+    val lineMappings = arrayListOf<RangeMapping>()
 
-    fun mapNewLineNumber(source: Int, currentIndex: Int, isLastMapped: Boolean, callSiteMarker: CallSiteMarker?): Int {
-        val dest: Int
-        val rangeMapping: RangeMapping
-        if (rangeMappings.isNotEmpty() && isLastMapped && couldFoldInRange(lastMappedWithNewIndex, source)) {
-            rangeMapping = rangeMappings.last()
-            rangeMapping.range += source - lastMappedWithNewIndex
-            dest = rangeMapping.mapSourceToDest(source)
+    fun mapNewLineNumber(source: Int, currentIndex: Int, callSiteMarker: CallSiteMarker?): Int {
+        var mapping = lineMappings.lastOrNull()
+        if (mapping != null && mapping.callSiteMarker == callSiteMarker &&
+            (source - mapping.source) in 0 until mapping.range + (if (mapping.maxDest == currentIndex) 10 else 0)
```

Comment:
In `couldFoldInRange` there was <= 10

Reply:
So `until` should be `..`
But I'm a bit confused by `mapping.maxDest == currentIndex '  check

Reply:
Yes, but `lastMappedWithNewIndex` was `mapping.source + mapping.range - 1`.

Reply:
The `mapping.maxDest == currentIndex` is equivalent to checking that this is the `FileMapping` that `mapNewLineNumber` was last called on. If it was not, then we cannot *extend* the range (because we've already used the next line number for some other file) but we can still reuse it (thus `0 until mapping.range` is OK -- but not `0..mapping.range`, because `mapping.source + mapping.range` is out of bounds for this mapping).

Reply:
Thanks! I've cleared my doubts